manufacturers/Conserwa/SensecapS2120/uplink_decoder.js

```diff
@@ -0,0 +1,185 @@
+		meas_list04.push(bytes[13]);
+		meas_list04.push((bytes[14] << 24) + (bytes[15] << 16) + (bytes[16] << 8) + bytes[17]);
+		meas_list04.push(bytes[18]/10);
+		meas_list04.push(((bytes[19]<<8) + bytes[20])/10);
+		meas_list04.push((bytes[22]<<8) + bytes[23]);
+		meas_list04.push(((bytes[24] << 24) + (bytes[25] << 16) + (bytes[26] << 8) + bytes[27])/1000);
+		meas_list04.push(((bytes[28]<<8) + bytes[29])*10);
+		meas_list04.push(bytes[1]);
+		meas_list04.push((bytes[6]<<8) + bytes[7]);
+		for(var i = 0; i<meas_list04.length; i++)
+		{
+			decoded.messages.push({
+                type: 'report_telemetry',
+				measurementId: i,
+                measurementValue: meas_list04[i]
+			});
+		}
+	}
+    // return
+    return {
+        data: decoded
+    };
+}
+function crc16Check(data) {
+    return true;
+}
+
+// util
+function bytes2HexString(arrBytes) {
+    var str = '';
+    for (var i = 0; i < arrBytes.length; i++) {
+        var tmp;
+        var num = arrBytes[i];
+        if (num < 0) {
+            tmp = (255 + num + 1).toString(16);
+        } else {
+            tmp = num.toString(16);
+        }
+        if (tmp.length === 1) {
+            tmp = '0' + tmp;
+        }
+        str += tmp;
+    }
+    return str;
+}
+
+function ngsildInstance(value, time, unit, dataset_suffix) {
+    var ngsild_instance = {
+        type: 'Property',
+        value: value,
+        observedAt: time
+    }
+    if (unit !== null) {
+        ngsild_instance.unitCode = unit
+    }
+    if (dataset_suffix !== null) {
+        ngsild_instance.datasetId = 'urn:ngsi-ld:Dataset:' + dataset_suffix
+    }
+    return ngsild_instance
+}
+
+function ngsildWrapper(input, time, entity_id) {
+    var ngsild_payload = [{
+        id: entity_id,
+        type: "Device"
+    }];
+    var messages = input.data.messages;
+    var error = true
+    for (let i = 0; i < messages.length; i++) {
+        if (messages[i].type === 'report_telemetry' && messages[i].measurementValue !== 0 && messages[i].measurementValue !== 2000001) {
+            error = false
+        }
+    }
+    if (error){
+        ngsild_payload[0].error = ngsildInstance(1, time, null, null)
+    } else {
+        for (let i = 0; i < messages.length; i++) {
+            if (messages[i].type === 'report_telemetry') {
+				if(messages[i].measurementId === 0) {
+					ngsild_payload[0].airTemperature = ngsildInstance(messages[i].measurementValue, time, 'CEL', 'Raw');
+				}
+                else if (messages[i].measurementId === 1) {
+					ngsild_payload[0].airHumidity = ngsildInstance(messages[i].measurementValue, time, 'P1', 'Raw');
+                }
+				else if (messages[i].measurementId === 2) {
+					ngsild_payload[0].lightIntensity = ngsildInstance(messages[i].measurementValue, time, 'LUX', 'Raw');
+                }
+				else if (messages[i].measurementId === 3) {
+					ngsild_payload[0].uvIndex = ngsildInstance(messages[i].measurementValue, time, '', 'Raw');
```

Comment:
If there is no unitCode, use `null` instead of `''`

Reply:
ok done